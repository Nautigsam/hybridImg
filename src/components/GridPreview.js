import React from 'react';
import { Link } from 'react-router-dom';
import { Box, Image, Heading, Media, Tag } from 'react-bulma-components';

import StoreProvider from '../store/StoreProvider';
export const GridPreview = ({ grid, hybrids }) => {
  const hybrid = hybrids[Math.floor(Math.random() * hybrids.length)];
  return (
    <Link to={`/grid/${grid.id}`}>
      <Box>
        <Heading size={4}>{grid.name}</Heading>
        <Media>
          <Media.Content>
            <figure className="image is-square is-96x96">
              <img src={hybrid.url} />
            </figure>
          </Media.Content>
          <Media.Content>
            <div>
              {grid.colThemes.map(tag => (
                <Tag key={tag.id} color="info">
                  {tag.name}
                </Tag>
              ))}
            </div>
            <div>
              {grid.lineThemes.map(tag => (
                <Tag key={tag.id} color="info" className="is-light">
                  {tag.name}
                </Tag>
              ))}
            </div>
          </Media.Content>
        </Media>
      </Box>
    </Link>
  );
};

function extraProps(store, props) {
  return {
    grid: store.getGrid(props.id),
    hybrids: store.getGridHybrids(props.id),
  };
}

export default StoreProvider(extraProps)(GridPreview);
