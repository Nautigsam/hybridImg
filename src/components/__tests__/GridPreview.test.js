import React from 'react';
import { StaticRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { GridPreview } from '../GridPreview';

describe('GridPreview', () => {
  it('renders correctly', () => {
    const props = {
      grid: {
        id: 'G1',
        user: 'U1',
        name: 'La grille de Mag',
        lineThemes: [
          { id: 'T1' },
          { id: 'T2' },
          { id: 'T3' },
          { id: 'T4' },
          { id: 'T5' },
          { id: 'T6' },
        ],
        colThemes: [
          { id: 'T7' },
          { id: 'T8' },
          { id: 'T9' },
          { id: 'T10' },
          { id: 'T11' },
          { id: 'T12' },
        ],
      },
      user: { id: 'u1', name: 'Mag' },
      users: [{ id: 'u1', name: 'Mag' }, { id: 'u2', name: 'Matt' }],
    };
    const tree = renderer
      .create(
        <StaticRouter>
          <GridPreview {...props} />
        </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
