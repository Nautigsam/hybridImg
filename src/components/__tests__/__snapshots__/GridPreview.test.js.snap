// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridPreview renders correctly 1`] = `
<a
  href="/grid/G1"
  onClick={[Function]}
>
  <div
    className="box"
  >
    <h1
      className="title is-4"
    >
      La grille de Mag
    </h1>
    <article
      className="media"
    >
      <div
        className="content"
      >
        <figure
          className="image is-square is-96x96"
        >
          <img
            src="./Images/20200411_112906.jpg"
          />
        </figure>
      </div>
      <div
        className="content"
      >
        <div>
          <span
            className="tag is-info"
          />
          <span
            className="tag is-info"
          />
          <span
            className="tag is-info"
          />
          <span
            className="tag is-info"
          />
          <span
            className="tag is-info"
          />
          <span
            className="tag is-info"
          />
        </div>
        <div>
          <span
            className="tag is-light is-info"
          />
          <span
            className="tag is-light is-info"
          />
          <span
            className="tag is-light is-info"
          />
          <span
            className="tag is-light is-info"
          />
          <span
            className="tag is-light is-info"
          />
          <span
            className="tag is-light is-info"
          />
        </div>
      </div>
    </article>
  </div>
</a>
`;
